modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -151,5 +165,18 @@ def enum_zones
       logger.debug "Enumerated authoritative dns zones: #{zones}"
       zones
     end
+
+    def enum_records(zone_name, node_name, type)
+      records = []
+      response = execute "/EnumRecords #{zone_name} #{node_name}.", "EnumRecords", true
+      response.each do |line|
+        logger.debug "/EnumRecords line=#{line}"
+        next unless line =~ /\s#{type}\s/
+        logger.debug "Extracting record from #{line}"
+        records << line.sub(/^.*#{type}\s+/, '').chomp("\n")
```

Comment:
I think this regex is not tight enough. You can either match zone/ttl/type/record and pick the last, or (which I think is easier) drop spaces from the end of the string and return a slice from the last space position to the end of the string.